android/build.gradle

```diff
@@ -57,7 +57,7 @@ class DefaultManifestPlaceHolders {
 }
 
 rootProject.childProjects.each { projName, proj ->
-    if (projName != 'app')
+    if (projName != 'app' && projName != 'react-native-onesignal')
```

Comment:
Apply defaults for manifest placeholders to the `react-native-onesignal` would remove the ability for them to be filled in by other targets such as the app. Since `react-native-onesignal` would be evaluated before `app`. More recent SDK setups have the OneSignal app_id in JS code however this change would break older setups as it would remove the app_id.

Instead we should apply defaults to the test project instead. This might be `appTests`, I recommend calling `println(projName)` to debug the issue

Reply:
Thanks for bringing that up 👍 Upon testing this change with `manifestPlaceholders`, I’ve seem to have stumbled over another issue. Could you take a look at https://github.com/geektimecoil/react-native-onesignal/pull/791?

Reply:
With https://github.com/geektimecoil/react-native-onesignal/pull/791 now being merged, I tested this again and everything worked fine. Are you sure this would break setups which specify the OneSignal app ID through manifest placeholders? Wouldn’t it working be explained by [the main manifest having a higher priority than the library’s manifest](https://developer.android.com/studio/build/manifest-merge#merge_priorities)?

Reply:
@sonicdoe You are right, I was thinking the `manifestPlaceholders` would be evaluated in a from a bottom up since it is using params from gradle but seems to follow the same for process for static manifest tags.

I tested removing the if `projName` check completely and didn't notice any issues in my testing. Removing this extra check means less code so I think this is the way to go.

Could you update your PR to remove the if completely? We can also remove the param to start with an underscore to not that we don't use it. `_projName`

Reply:
If we were to completely remove the check, we would inject manifest placeholders into _all_ child projects, right? Of course, with manifest merger’s priorities it should still result in the same manifest but I’m wary of react-native-onesignal injecting things into other (third-party) projects. It might work for now but it could have unintended consequences later on. What do you think?

Reply:
Good point, it would add these manifestPlaceHolders to all projects which wouldn't be ideal. Since they are unique it should be fine however it would be rather confusing to see these everywhere if someone was debugging an issue.

So in the end I think your PR is good as you have it, I'll mark this as resolved.

Thanks for the attention to detail!